### DISCUSSION TITLE
Creating more than 3 lines with different styles in ChartView programatically

### DISCUSSION THREAD

**Discussion by JayantSharma777** [patrykandpatrick/vico#399] [Questions] [answered]
I am facing problem in adding more than 3 lineSpecs so that my chartView which is of line type can have more than 3 different lines.I can not do this programatically because i do not get the option of adding lines in the chartView object.

**Module**

- [ ] `compose`
- [x] `views`
- [ ] `compose-m2`
- [ ] `compose-m3`
- [ ] `core` (select if none of the above apply)

**To reproduce**
set up a chartView in xml .
Initialize the chartView object in your activity and link it with your xml,

**Expected behaviour**
I want to add more than 3 different lines in my chartView of line type.

**Android versions**
12,13

**Comment by patrickmichalik** [comment, marked as answer]
Hello! `ChartView` is a host component. It doesn’t implement `Chart`. You can add `LineSpec`s programmatically by using `(chartView.chart as LineChart).lines`, where `chartView` is a reference to your `ChartView` instance.

**Reply by JayantSharma777** [reply]
oh Okay, Thanks
It is working now

**Reply by patrickmichalik** [reply]
You’re welcome!

**Reply by JayantSharma777** [reply]
Now in this line chart, I am trying to add styles for my custom line, and i want to add a drawable for my point and change it's style, but in you implementation the point is of type Component which is an abstract class, how can i extend it or use it,I have asked you this before as well and you told me to do it programatically then too but i couldn't find a way. can you make that clear as well?
<img width="717" alt="image" src="https://github.com/patrykandpatrick/vico/assets/75779183/702f7ff4-9c17-4649-82cc-c52ad20c53f1">
This is how i am adding the lineSpec in lines in ChartView

**Reply by patrickmichalik** [reply]
Please use `ShapeComponent` and `Shapes.drawableShape`: `ShapeComponent(Shapes.drawableShape(...), ...)`.
